src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1530,6 +1544,107 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
+    
+    deployment_storage_auth_type = deployment_storage_auth_type or 'systemAssignedIdentity'
+
+    if deployment_storage_auth_value and deployment_storage_auth_type != 'userAssignedIdentity':
+        raise ArgumentUsageError(
+            '--deployment-storage-auth-value is only a valid input for --deployment-storage-auth-type set to userAssignedIdentity. '
+            'Please try again with --deployment-storage-auth-type set to userAssignedIdentity.'
+        )
+
+    if deployment_storage_auth_type == 'userAssignedIdentity':
+        deployment_storage_user_assigned_identity = _get_or_create_user_assigned_identity(cmd, resource_group_name, name, deployment_storage_auth_value, functionapp.location)
+        deployment_storage_auth_value = deployment_storage_user_assigned_identity.id
+    elif deployment_storage_auth_type == 'storageAccountConnectionString':
+        logger.warning("deployment_storage_auth_type is %s", deployment_storage_auth_type)
+        deployment_storage_conn_string = _get_storage_connection_string(cmd.cli_ctx, deployment_storage)
+        configs = get_site_configs(cmd, resource_group_name, name, slot)
+        if configs.app_settings is None:
+            configs.app_settings = []
+        configs.app_settings.append(NameValuePair(name='DEPLOYMENT_STORAGE_CONNECTION_STRING',
+                                                        value=deployment_storage_conn_string))
+        update_flex_functionapp_configuration(cmd, resource_group_name, name, configs)
+        deployment_storage_auth_value = 'DEPLOYMENT_STORAGE_CONNECTION_STRING'
+        logger.warning("deployment_storage_auth_type is %s", deployment_storage_auth_type)
+
+    # if (functionapp_config is None):
+    functionapp_config = {} #TODO: replace with actual model when api version is relesed
+
+    if (functionapp_config.get('deployment') is None):
+        functionapp_config['deployment'] = {} #TODO: replace with the model if needed.
+
+    if (functionapp_config['deployment'].get('storage') is None):
+        functionapp_config['deployment']['storage'] = {} #TODO: replace with the model if needed.
+
+    
+    if (deployment_config_storage_value is None):
+        functionapp_config['deployment']['storage']['type'] = 'blobContainer'
+        functionapp_config['deployment']['storage']['value'] = deployment_config_storage_value
+        
+    functionapp_config['deployment']['storage']['authentication'] = {}
+    functionapp_config['deployment']['storage']['authentication']['type'] = deployment_storage_auth_type
+    functionapp_config['deployment']['storage']['authentication']['value'] = deployment_storage_auth_value
+
+    if deployment_storage_auth_type == 'userAssignedIdentity':
+        assign_identities = [deployment_storage_auth_value]
+        _assign_deployment_storage_managed_identity_role(cmd.cli_ctx, deployment_storage, deployment_storage_user_assigned_identity.principal_id)
```

Comment:
Are we also expecting the client tools to assign the roles? We might need to confirm this with PMs